InaraResponder/InaraResponder.cs

```diff
@@ -534,58 +534,64 @@ private void handleShipTransferInitiatedEvent(ShipTransferInitiatedEvent @event)
 
         private void handleShipRenamedEvent(ShipRenamedEvent @event)
         {
-            Ship currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
-            Dictionary<string, object> currentShipData = new Dictionary<string, object>()
-            {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+            var currentShipData = new Dictionary<string, object>()
+            {
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
+                { "shipName", @event.name },
+                { "shipIdent", @event.ident },
                 { "isCurrentShip", true }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipRole", (currentShip?.Role ?? Role.MultiPurpose).invariantName);
+                currentShipData.Add("isHot", currentShip?.hot ?? false);
+            }
             inaraService.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "setCommanderShip", currentShipData));
         }
 
         private void handleShipLoadoutEvent(ShipLoadoutEvent @event)
         {
-            Ship currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
-            Dictionary<string, object> currentShipData = new Dictionary<string, object>()
-            {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+            var currentShipData = new Dictionary<string, object>()
+            {
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
+                { "shipName", @event.shipname },
+                { "shipIdent", @event.shipident },
                 { "isCurrentShip", true },
                 { "shipHullValue", @event.hullvalue },
                 { "shipModulesValue", @event.modulesvalue },
                 { "shipRebuyCost", @event.rebuy }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipRole", (currentShip?.Role ?? Role.MultiPurpose).invariantName);
```

Comment:
We can remove the `?` from `currentShip`.

Reply:
No. The `GetShip` method can return null a null value if the local ship id is null or not found int he shipyard.